extensions/appdynamics/extension.py

```diff
@@ -0,0 +1,202 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""AppDynamics Extension
+Downloads, installs and configures the AppDynamics agent for PHP
+"""
+import os
+import os.path
+import logging
+
+_log = logging.getLogger('appdynamics')
+
+DEFAULTS = {
+'APPDYNAMICS_HOST': 'packages.appdynamics.com',
+'APPDYNAMICS_VERSION': '4.1.5.0',
+'APPDYNAMICS_PACKAGE': 'appdynamics-php-agent-x64-linux-{APPDYNAMICS_VERSION}.tar.bz2',
+'APPDYNAMICS_DOWNLOAD_URL': 'https://{APPDYNAMICS_HOST}/'
+                         'php/{APPDYNAMICS_VERSION}/{APPDYNAMICS_PACKAGE}',
+}
+
+class AppDynamicsInstaller(object):
```

Comment:
New extensions should use [ExtensionHelper](https://github.com/cloudfoundry/php-buildpack/blob/master/lib/extension_helpers.py#L19) or [PHPExtensionHelper](https://github.com/cloudfoundry/php-buildpack/blob/master/lib/extension_helpers.py#L158).  This removes a lot of the boiler plate code in your extension.  See the [session extension](https://github.com/cloudfoundry/php-buildpack/blob/master/extensions/sessions/extension.py#L74) for an example.